core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -539,6 +540,31 @@ public static String getUser(String authorization, ContainerRequestContext reque
     }
   }
 
+  public static String getUser(String authorization, HttpServletRequest request)
+          throws S3Exception {
+    if (S3RestUtils.isAuthenticationEnabled(Configuration.global())) {
+      return getUserFromSignature(request);
+    }
+    try {
+      return getUserFromAuthorization(authorization, Configuration.global());
```

Comment:
what do u mean? are u saying these even getUserFromAuthorization does the logic for generating strToSign it doesnt actually do the comparing with the passed-in ones to do actual auth, and we should just use  getUserFromSignature only instead of extracting the user from header by the func getUserFromAuthorization?

Reply:
we start with the `getUserFromAuthorization`, and later the community contributor contributes the `getUserFromSignature`. They are almost the same function and can do the same thing. The only problem is the new one need more test so we have a switch to enable it.